analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TestAnalysisComparer_2.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class SecondClass
+    {
+        async void WrongLength() { } // Noncompliant {{Return 'Task' instead.}}
+        //    ^^^^^
+
+        void Default()// Secondary [flow-0]
+                      // Secondary@-1 [flow-1] {{Wrong message}}
```

Comment:
_More cases:_ Secondary can also have `^^^^^` markings on the secondary location (that can be correct or wrong)

Reply:
There is a problem with how secondary locations ids are parsed in some of the cases so I will add this case in a following PR